thirdeye/thirdeye-frontend/app/pods/components/rootcause-anomaly/component.js

```diff
@@ -0,0 +1,74 @@
+import Ember from 'ember';
+import moment from 'moment';
+
+export default Ember.Component.extend({
+  // TODO implement anomaly feedback
+
+  entities: null, // {}
+
+  anomalyUrn: null, // ""
+
+  anomaly: Ember.computed(
+    'entities',
+    'anomalyUrn',
+    function () {
+      const { entities, anomalyUrn } = this.getProperties('entities', 'anomalyUrn');
+
+      if (!anomalyUrn || !entities || !entities[anomalyUrn]) { return false; }
+
+      return entities[anomalyUrn];
+    }
+  ),
+
+  functionName: Ember.computed('anomaly', function () {
+    return this.get('anomaly').attributes.function[0];
+  }),
+
+  anomalyId: Ember.computed('anomaly', function () {
+    return this.get('anomaly').urn.split(':')[3];
+  }),
+
+  metric: Ember.computed('anomaly', function () {
+    return this.get('anomaly').attributes.metric[0];
+  }),
+
+  dataset: Ember.computed('anomaly', function () {
+    return this.get('anomaly').attributes.dataset[0];
+  }),
+
+  current: Ember.computed('anomaly', function () {
+    return parseFloat(this.get('anomaly').attributes.current[0]);
+  }),
+
+  baseline: Ember.computed('anomaly', function () {
+    return parseFloat(this.get('anomaly').attributes.baseline[0]);
+  }),
+
+  change: Ember.computed('anomaly', function () {
+    const attr = this.get('anomaly').attributes;
+    return parseFloat(attr.current[0]) / parseFloat(attr.baseline[0]) - 1;
+  }),
+
+  status: Ember.computed('anomaly', function () {
+    return this.get('anomaly').attributes.status[0];
+  }),
+
+  duration: Ember.computed('anomaly', function () {
+    const anomaly = this.get('anomaly');
+    return moment.duration(anomaly.end - anomaly.start).humanize();
+  }),
+
+  dimensions: Ember.computed('anomaly', function () {
+    const attr = this.get('anomaly').attributes;
+    const dimNames = attr.dimensions;
+    const dimValues = dimNames.reduce((agg, n) => { agg[n] = attr[n][0]; return agg; }, {});
```

Comment:
nitpick: let's have better naming than 'n' for readability's sake

Reply:
fixed